src/validationUtils.ts

```diff
@@ -66,3 +66,32 @@ export function validateAMKA(amka: string | number): boolean {
   // The sum should be divisible by 10
   return sum % 10 === 0;
 }
+
+/**
+ * Validates a VAT (Value Added Tax) number based on the following criteria:
+ * - Must be exactly 9 digits long.
+ * - Must consist of numeric characters only.
+ * - Cannot be all zeros.
+ * - The last digit (checksum) must satisfy a specific calculation based on the preceding digits.
+ *
+ * @param {string | number} vatNumber - The VAT number to validate, provided as a string or a number.
+ * @returns {boolean} - Returns `true` if the VAT number is valid according to the specified rules; otherwise, `false`.
+ */
+export function validateVATNumber(vatNumber: string | number): boolean {
+  vatNumber = String(vatNumber);
+
+  if (vatNumber.length != 9 || !/^\d+$/.test(vatNumber) || vatNumber === "0".repeat(9)) {
+    return false;
+  }
+
+  const sum = vatNumber
+    .substring(0, 8)
+    .split("")
+    .reduce((s, v, i) => s + (parseInt(v) << (8 - i)), 0);
+
+  const calc = sum % 11;
+  const d9 = parseInt(vatNumber[8]!);
+  const valid = calc % 10 === d9;
+
+  return valid;
+}
```

Comment:
```suggestion
export function validateVATNumber(vatNumberInput: string | number): boolean {
  const vatNumber = String(vatNumberInput);
 
  if (vatNumber.length !== 9 || !/^\d+$/.test(vatNumber) || vatNumber === "0".repeat(9)) {
    return false;
  }

  const sum = vatNumber
    .substring(0, 8)
    .split("")
    .reduce((s, v, i) => s + (parseInt(v) << (8 - i)), 0);

  const calc = sum % 11;
  const d9 = parseInt(vatNumber[8]);
  const valid = calc % 10 === d9;

  return valid;
}
```

Reply:
TS was complaining when removing ```!``` from ```vatNumber[8]!``` thus I kept it.